rust/operator/src/lib.rs

```diff
@@ -0,0 +1,831 @@
+    /// The 'zoo.cfg' properties are read from the product_config and/or merged with the cluster
+    /// custom resource.
+    ///
+    /// Labels are automatically adapted from the `recommended_labels` with a type (data for
+    /// 'zoo.cfg' and id for 'myid'). Names are generated via `name_utils::build_resource_name`.
+    ///
+    /// Returns a map with a 'type' identifier (e.g. data, id) as key and the corresponding
+    /// ConfigMap as value. This is required to set the volume mounts in the pod later on.
+    ///
+    /// # Arguments
+    ///
+    /// - `pod_id` - The `PodIdentity` containing app, instance, role, group names and the id.
+    /// - `validated_config` - The validated product config.
+    /// - `id_mapping` - All id to node mappings required to create config maps
+    ///
+    async fn create_config_maps(
+        &self,
+        pod_id: &PodIdentity,
+        role: &DruidRole,
+        validated_config: &HashMap<PropertyNameKind, BTreeMap<String, String>>,
+    ) -> Result<HashMap<&'static str, ConfigMap>, Error> {
+        let mut config_maps = HashMap::new();
+        let mut cm_conf_data = BTreeMap::new();
+
+        for (property_name_kind, config) in validated_config {
+            let mut transformed_config: BTreeMap<String, Option<String>> = config
+                .iter()
+                .map(|(k, v)| (k.clone(), Some(v.clone())))
+                .collect();
+
+            match property_name_kind {
+                PropertyNameKind::File(file_name) if file_name == RUNTIME_PROPS => {
+                    // NOTE: druid.host can be set manually - if it isn't, the canonical host name of
+                    // the local host is used.  This should work with the agent and k8s host networking
+                    // but might need to be revisited in the future
+
+                    if let Some(zk_info) = &self.zookeeper_info {
+                        transformed_config.insert(
+                            ZOOKEEPER_CONNECTION_STRING.to_string(),
+                            // Some("172.17.0.4:2181".to_string()),
+                            Some(zk_info.connection_string.clone()),
+                        );
+                    } else {
+                        return Err(error::Error::ZookeeperConnectionInformationError);
+                    }
+
+                    let runtime_properties = get_runtime_properties(role, &transformed_config);
+                    cm_conf_data.insert(RUNTIME_PROPS.to_string(), runtime_properties);
+                }
+                PropertyNameKind::File(file_name) if file_name == JVM_CONFIG => {
+                    let jvm_config = get_jvm_config(role);
+                    cm_conf_data.insert(JVM_CONFIG.to_string(), jvm_config);
+                }
+                PropertyNameKind::File(file_name) if file_name == LOG4J2_CONFIG => {
+                    let log_config = get_log4j_config(role);
+                    cm_conf_data.insert(LOG4J2_CONFIG.to_string(), log_config);
```

Comment:
The way this is build we never actually use the config overrides. 
Please check e.g. https://github.com/stackabletech/hdfs-operator/blob/aab7c8b190af3700656c4996ad89c2cbba63b7a8/rust/operator/src/lib.rs#L293

Reply:
I see, I'm working on it

Reply:
Okay, I think for the runtime properties the transformed_config is already used, so that should work right? And the other two are not templated, so it's not possible to implement there.